PAWS/ECGRecordings/ECGModule.swift

```diff
@@ -96,4 +96,133 @@ class ECGModule: Module, DefaultInitializable, EnvironmentAccessible {
+                    let content = UNMutableNotificationContent()
+                    content.title = "Upload Error"
+                    content.body = "Sample could not be uploaded \(supplementalMetric.sampleType.description) (\(supplementalMetric.uuid.uuidString) at \(Date.now.formatted(date: .numeric, time: .complete)): \((supplementalMetric as? HKQuantitySample)?.quantity.description ?? "Unknown")"
+                    let request = UNNotificationRequest(identifier: UUID().uuidString, content: content, trigger: nil)
+                    try? await UNUserNotificationCenter.current().add(request)
+                }
+            }
+        } catch {
+            logger.log("Could not access HealthKit sample: \(error)")
+            await addECGMessage(for: electrocardiogram)
+        }
+    }
+    
+    
+    /// Creates a notification with a title and body message when there is an error accessing a HealthKit sample.
+    /// - Parameter electrocardiogram: The `HKElectrocardiogram` object for which the error occurred.
+    func addECGMessage(for electrocardiogram: HKElectrocardiogram) async {
+        let content = UNMutableNotificationContent()
+        content.title = "HealthKit Error"
+        content.body = "Sample \(electrocardiogram.sampleType.description) with identifier \(electrocardiogram.uuid.uuidString)"
+        let request = UNNotificationRequest(identifier: UUID().uuidString, content: content, trigger: nil)
+        try? await UNUserNotificationCenter.current().add(request)
+    }
+    
+    private func uploadUnuploadedECGs() async throws {
+        for ecg in electrocardiograms where try await !isUploaded(ecg) {
+            Task {
+                do {
+                    try await self.upload(sample: ecg)
+                } catch {
+                    logger.log("Could not access HealthKit sample: \(error)")
+                    await addECGMessage(for: ecg)
+                }
+            }
+        }
+    }
+    
+    /// Reloads the ECGs by checking if the user is authenticated.
+    /// If the user is authenticated, it sets a sample predicate for the HealthKit query based on the user's account creation date and the current date.
+    /// - Throws: An error if the user is not authenticated.
+    func reloadECGs() async throws {
+        guard let user = Auth.auth().currentUser else {
+            logger.error("User not authenticated")
+            return
+        }
+
+        let samplePredicate = HKQuery.predicateForSamples(withStart: user.metadata.creationDate, end: .now, options: .strictStartDate)
+        let queryDescriptor = HKSampleQueryDescriptor(
+            predicates: [HKSamplePredicate<HKElectrocardiogram>.electrocardiogram(samplePredicate)],
+            sortDescriptors: []
+        )
+        let samples = try await queryDescriptor.result(for: healthStore)
+        
+        self.electrocardiograms = samples.filter { !self.electrocardiograms.contains($0) }
```

Comment:
We should either completely replace the list of ECGs or add a smart diffing here (we already have an add ECG function). I would lean to just replace them based on the query result. The current code removes all existing ones and only adds the ones that have not been there which is not the intended behaviour.